### PR TITLE
Many client updates

### LINKED ISSUE #8455: systemd-resolve should also show the global settings about LLMNR or MDNS
### Submission type

 - Bug report

### systemd version the issue has been seen with

> 238

### Used distribution

> Linux From Scratch

### In case of bug report: Expected behaviour you didn't see

> Disabling LLMNR shows in `systemd-resolve --status`.

### In case of bug report: Unexpected behaviour you saw

> With the following configuration, LLMNR is still shown as enabled.

```
$ more /etc/systemd/resolved.conf.d/disable-llmnr.conf
[Resolve]
LLMNR=no
$ systemd-resolve --status
Global
         DNS Servers: 141.14.16.1
          DNS Domain: molgen.mpg.de
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
                      17.172.in-addr.arpa
                      18.172.in-addr.arpa
                      19.172.in-addr.arpa
                      20.172.in-addr.arpa
                      21.172.in-addr.arpa
                      22.172.in-addr.arpa
                      23.172.in-addr.arpa
                      24.172.in-addr.arpa
                      25.172.in-addr.arpa
                      26.172.in-addr.arpa
                      27.172.in-addr.arpa
                      28.172.in-addr.arpa
                      29.172.in-addr.arpa
                      30.172.in-addr.arpa
                      31.172.in-addr.arpa
                      corp
                      d.f.ip6.arpa
                      home
                      internal
                      intranet
                      lan
                      local
                      private
                      test

Link 3 (net00)
      Current Scopes: none
       LLMNR setting: yes
MulticastDNS setting: no
      DNSSEC setting: yes
    DNSSEC supported: yes

Link 2 (dummy0)
      Current Scopes: none
       LLMNR setting: yes
MulticastDNS setting: no
      DNSSEC setting: yes
    DNSSEC supported: yes
```

### In case of bug report: Steps to reproduce the problem

> Configure systemd-resolved as above.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #8455.

### CODE REVIEW

**Comment by yuwata** [comment]
I've merged and updated #8426 and #8462, as they conflict each other.

File: src/busctl/busctl.c [hunk 1]
```diff
@@ -1978,46 +2054,26 @@ static int parse_argv(int argc, char *argv[]) {
...
+                { "list",         VERB_ANY, 1,        VERB_DEFAULT, list_bus_names },
+                { "status",       VERB_ANY, 2,        0,            status         },
+                { "monitor",      VERB_ANY, VERB_ANY, 0,            verb_monitor   },
+                { "capture",      VERB_ANY, VERB_ANY, 0,            verb_capture   },
+                { "tree",         VERB_ANY, VERB_ANY, 0,            tree           },
+                { "introspect",   3,        4,        0,            introspect     },
+                { "call",         5,        VERB_ANY, 0,            call           },
+                { "get-property", 5,        VERB_ANY, 0,            get_property   },
+                { "set-property", 6,        VERB_ANY, 0,            set_property   },
+                { "help",         VERB_ANY, VERB_ANY, 0,            verb_help      },
```
Comment (keszybz) [inline src/busctl/busctl.c:2070]: Yep, those all seem to work fine now.

File: src/hostname/hostnamectl.c [hunk 2]
```diff
@@ -163,41 +163,29 @@ static int show_all_names(sd_bus *bus, sd_bus_error *error) {
...
-        free(info.chassis);
-        free(info.deployment);
-        free(info.location);
-        free(info.kernel_name);
-        free(info.kernel_release);
-        free(info.os_pretty_name);
-        free(info.os_cpe_name);
-        free(info.virtualization);
-        free(info.architecture);
-
```
Comment (keszybz) [inline src/hostname/hostnamectl.c:199]: Nice.

File: src/shared/bus-util.c [hunk 3]
```diff
@@ -1060,13 +1060,13 @@ static int map_basic(sd_bus *bus, const char *member, sd_bus_message *m, sd_bus_
 
         case SD_BUS_TYPE_BOOLEAN: {
                 unsigned b;
-                int *p = userdata;
+                bool *p = userdata;
```
Comment (poettering) [inline src/shared/bus-util.c:1104]: uh, this one is problematic, i am not sure we should do that. Currently, we never expose the bool type in our public APIs, as we avoid too many C99isms in that. We only use it internally. That means the whole of sd-bus uses "int" for booleans in its public API (though "bool" internally wherever applicable). Now, src/shared/bus-util.c is not really part of sd-bus, and is internally, but the already existing pitfalls of using different types for booleans is increased I think if bus-util.c doesn't follow sd-bus in this regard.

I also have the suspicion that there might be various users already that expect bools to be ints which will trip up on this type change. 

I am pretty sure we should find a different solutoin for this, for example, by providing an explicit setter function for bools here, i.e. add bus_map_bool() similar to bus_map_id128(), which does the necessary conversion to bool.

**Comment by poettering** [comment]
BTW, @yuwata awesome work, thank you a million for these kind of clean-ups, this is *very* welcome!

File: src/shared/bus-util.c (hunk 3 above)
Comment (yuwata) [reply src/shared/bus-util.c:1104]: Hmm. I see. But I do not want to specify `bus_map_bool()` to so many properties. I've tried to support this problem in another way in #8546. Please take a look.

**Pull request merged**